File: Java/AerisDemo/src/com/xweather/sdkdemo/java/db/MyPlace.java
package com.xweather.sdkdemo.java.db;

import com.aerisweather.aeris.util.WeatherUtil;

public class MyPlace {

	public String name;
	public String country;
	public String state;
	public boolean myLoc = false;
	public double latitude;
	public double longitude;

	public String getTextDisplay(String defaultText) {
		String text = defaultText;
		if (state != null && state.length() > 0) {
			text = String.format("%s, %s, %s", WeatherUtil.capitalize(name),
					state.toUpperCase(), country.toUpperCase());
		} else {
			text = String.format("%s, %s", WeatherUtil.capitalize(name),
					country.toUpperCase());
		}
		return text;
	}

}

File: Kotlin/AerisSdkDemo/app/src/main/java/com/xweather/sdkdemo/kotlin/data/room/MyPlace.kt
package com.xweather.sdkdemo.kotlin.data.room

import androidx.room.Entity
import com.aerisweather.aeris.util.WeatherUtil
import com.google.android.gms.maps.model.LatLng
import java.util.*

@Entity(tableName = "my_place_table", primaryKeys = ["latitude", "longitude"])
class MyPlace(
    var myLoc: Boolean = false,
    var latitude: Double = 0.0,
    var longitude: Double = 0.0,
    var name: String? = "unknown",
    var state: String? = "unknown",
    var country: String? = "unknown",
) {

    fun getTextDisplay(defaultText: String?): String? {
        return if (defaultText.isNullOrEmpty()) {
            String.format(
                "%s, %s",
                WeatherUtil.capitalize(name),
                country?.uppercase(Locale.getDefault()),
            )
        } else {
            String.format(
                "%s, %s, %s",
                WeatherUtil.capitalize(name),
                state?.uppercase(Locale.getDefault()),
                country?.uppercase(Locale.getDefault()),
            )
        }
    }

    fun getLatLong() = LatLng(latitude, longitude)
}
